### PR TITLE
feat: CRM Settings

### PR DESCRIPTION
**Pull request by anupamvs**
Docs- https://docs.erpnext.com/docs/v13/user/manual/en/CRM/crm_settings
- Moved out some settings from the Selling (Campaign Naming By, Default Quotation Validity Days, Close Opportunity After Days)

image
**New Settings**
- Auto Creation of Contact for new Lead
- Allowed users to create multiple leads from the same email
- User can toggle the creation of Event on Next Contact Date for both Lead and Opportunity.

### CODE REVIEW

**Comment by umairsy** [comment]
1. Introduce Column break in Lead and Opportunity section, to maintain left - right balance.
2.  Move first section in the last and put section heading. May be third section can be called "Quotation Settings", and out campaign setting under Leads.

File: erpnext/selling/doctype/crm_settings/crm_settings.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  }
+ ],
+ "icon": "fa fa-cog",
+ "index_web_pages_for_search": 1,
+ "issingle": 1,
+ "links": [],
+ "modified": "2021-10-04 15:42:30.981243",
+ "modified_by": "Administrator",
+ "module": "Selling",
+ "name": "CRM Settings",
```
Comment (ruchamahabal) [inline erpnext/selling/doctype/crm_settings/crm_settings.json]: We need a patch to update CRM Settings for Selling Settings.

File: erpnext/selling/doctype/crm_settings/crm_settings.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  {
+   "fieldname": "section_break_5",
+   "fieldtype": "Section Break",
+   "label": "Lead"
+  },
+  {
+   "default": "0",
+   "fieldname": "allow_creation_of_leads_form_a_email",
+   "fieldtype": "Check",
+   "label": "Allow Creation of Multiple Leads form a Email"
```
Comment (ruchamahabal) [inline erpnext/selling/doctype/crm_settings/crm_settings.json]:
```suggestion
   "label": "Allow Creation of Multiple Leads from an Email"
```

**Allow Lead Duplication based on Emails** 

Is this better?

File: erpnext/selling/doctype/crm_settings/crm_settings.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+   "fieldtype": "Data",
+   "label": "Default Quotation Validity Days"
+  },
+  {
+   "fieldname": "column_break_9",
+   "fieldtype": "Column Break"
+  },
+  {
+   "default": "15",
+   "description": "Auto close Opportunity after the no. of days mentioned above",
```
Comment (ruchamahabal) [inline erpnext/selling/doctype/crm_settings/crm_settings.json]:
```suggestion
   "description": "Auto close Replied Opportunity after the no. of days mentioned above",
```

File: erpnext/selling/doctype/crm_settings/crm_settings.json [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+   "fieldname": "close_opportunity_after_days",
+   "fieldtype": "Int",
+   "label": "Close Opportunity After Days"
```
Comment (ruchamahabal) [inline erpnext/selling/doctype/crm_settings/crm_settings.json]:
```suggestion
   "label": "Close Replied Opportunity After Days"
```

File: erpnext/crm/doctype/opportunity/opportunity.py [hunk 5]
```diff
@@ -71,19 +70,20 @@ def make_new_lead_if_required(self):
 		"""Set lead against new opportunity"""
 		if (not self.get("party_name")) and self.contact_email:
 			# check if customer is already created agains the self.contact_email
-			customer = frappe.db.sql("""select
-				distinct `tabDynamic Link`.link_name as customer
-				from
+			customer = frappe.db.sql("""SELECT
```
Comment (ruchamahabal) [inline erpnext/crm/doctype/opportunity/opportunity.py]: Can be rewritten using `frappe.qb`?

**Review by ruchamahabal — CHANGES_REQUESTED** [review]

File: erpnext/crm/doctype/opportunity/opportunity.py (hunk 5 above)
Comment (anupamvs) [reply erpnext/crm/doctype/opportunity/opportunity.py]: I don't think so join is possible.

Comment (ruchamahabal) [reply erpnext/crm/doctype/opportunity/opportunity.py]: @anupamvs join is possible

image

You can convert all the queries and fix linter issues

**Review by ruchamahabal — APPROVED** [review]

**Pull request merged**